src/run/check_system.rs

```diff
@@ -0,0 +1,77 @@
+use std::process::Command;
+
+use serde::{Deserialize, Serialize};
+use sysinfo::System;
+
+use crate::prelude::*;
+
+fn get_user() -> Result<String> {
+    let user_output = Command::new("whoami")
+        .output()
+        .map_err(|_| anyhow!("Failed to get user info"))?;
+    if !user_output.status.success() {
+        bail!("Failed to get user info");
+    }
+    let output_str =
+        String::from_utf8(user_output.stdout).map_err(|_| anyhow!("Failed to parse user info"))?;
+    Ok(output_str.trim().to_string())
+}
+
+#[derive(Eq, PartialEq, Hash, Serialize, Deserialize, Debug, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct SystemInfo {
+    pub os: String,
+    pub os_version: String,
+    pub arch: String,
+    pub host: String,
+    pub user: String,
+}
+
+#[cfg(test)]
+impl SystemInfo {
+    pub fn test() -> Self {
+        SystemInfo {
+            os: "Ubuntu".to_string(),
+            os_version: "20.04".to_string(),
+            arch: "x86_64".to_string(),
+            host: "host".to_string(),
+            user: "user".to_string(),
+        }
+    }
+}
+
+/// Checks if the system is supported and returns the system info
+///
+/// Supported systems:
+/// - Ubuntu 20.04 on x86_64
+/// - Ubuntu 22.04 on x86_64
+/// - Debian 11 on x86_64
+/// - Debian 12 on x86_64
+pub fn check_system() -> Result<SystemInfo> {
```

Comment:
Let's split this in two methods: get and check. Doing both here works, but we lose the valuable checks among all the error-handling